bv-java-grupa-1/iohomework/dataprocessing/ProcessCountryData.java

```diff
@@ -0,0 +1,108 @@
+package dataprocessing;
+
+import customexception.CountryNotFoundException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+
+/**
+ * Deals with all the data processing from the online document.
+ */
+public class ProcessCountryData {
+    // Stores a list with country objects.
+    private List<CountryData> countryList = new ArrayList<>();
+
+    public ProcessCountryData() {
+        addCountries();
+    }
+
+    /**
+     * @return a list with every line from the online document.
+     */
+    public List<String> getURLData() {
+        List<String> linesList = new ArrayList<>();
+
+        try {
+            URL url = new URL("https://openei.org/doe-opendata/dataset/a7fea769-691d-4536-8ed3-471e993a2445/resource/86c50aa8-e40f-4859-b52e-29bb10166456/download/populationbycountry19802010millions.csv");
+            BufferedReader in = new BufferedReader(new InputStreamReader(url.openStream()));
+            String line;
+
+            while ((line = in.readLine()) != null ) {
+                linesList.add(line);
+            }
+            in.close();
+        } catch (MalformedURLException e) {
+            System.out.println("Malformed URL: " + e.getMessage());
+        } catch (IOException e) {
+            System.out.println("I/O Error: " + e.getMessage());
+        }
+        return linesList;
+    }
+
+    /**
+     * Takes the data retrieved from the online document and organizes it in a list of Objects, each representing
+     * a different country.
+     */
+    public void addCountries() {
```

Comment:
Same as above; since it's called from the constructor, I don't think anybody else needs to "see" it.